teamcity: failed tests on master: acceptance/TestMonotonicInserts

The following tests appear to have failed:

[#44249](https://teamcity.cockroachdb.com/viewLog.html?buildId=44249):

```
--- FAIL: acceptance/TestMonotonicInserts (48.215s)
	testing.go:117: acceptance/util_test.go:404, condition failed to evaluate within 45s: failed to send RPC: rpc error: code = 14 desc = grpc: the connection is unavailable
	localcluster.go:699: unexpected extra event &{0 die} (after [])

```
Please assign, take a look and update the issue accordingly.

Another here: https://teamcity.cockroachdb.com/viewLog.html?buildId=45356&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_AcceptanceProposerEvaluatedKv

Logs suggest that nodes 0 and 1 crashed with nil pointer panic, not sure if that's the underlying issue or something that just happens in acceptance tests that fail unexpectedly.

```I161110 21:08:11.949550 215 rpc/context.go:220  [n1] dialing roach-27e4c2e6-2:26257
I161110 21:08:11.958550 281 rpc/clock_offset.go:168  [n1] 3 of 3 nodes are within the maximum offset of 250ms
I161110 21:08:11.961384 215 storage/store.go:3126  [n1,replicate,s1,r1/1:/{Min-Table/11}] streamed snapshot: kv pairs: 596, log entries: 13
I161110 21:08:11.966358 215 storage/replica_command.go:3244  [n1,replicate,s1,r1/1:/{Min-Table/11}] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I161110 21:08:11.968265 215 storage/log.go:70  [n1,replicate,s1,r1/1:/{Min-Table/11}] Range Event: "add", range: 1, info: {"AddReplica":{"node_id":3,"store_id":3,"replica_id":3},"UpdatedDesc":{"range_id":1,"end_key":"kw==","replicas":[{"node_id":1,"store_id":1,"replica_id":1},{"node_id":2,"store_id":2,"replica_id":2},{"node_id":3,"store_id":3,"replica_id":3}],"next_replica_id":4}}
I161110 21:08:11.971199 215 sql/planner.go:238  [log-range-event] internal-txn, memory usage max 1.4 KiB
I161110 21:08:11.971220 215 sql/planner.go:239  [log-range-event] internal-session, memory usage max 0 B
I161110 21:08:11.971227 215 sql/planner.go:240  [log-range-event] internal-root, memory usage max 10 KiB
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xabaf79]

goroutine 114 [running]:
panic(0x169ae60, 0xc42001a060)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc4203ecfc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0x6e
panic(0x169ae60, 0xc42001a060)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).PushTxn(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0x7f7f20834e38, 0xc420d30e00, 0xc420dec700, 0x1485cbc33770374b, 0x0, 0x100000001, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:1417 +0x819
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeCmd(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0xc420b18510, 0x8, 0x0, 0x7f7f20834e38, 0xc420d30e00, 0xc420dec700, 0x1485cbc33770374b, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:144 +0x2b4c
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeBatch(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0xc420b18510, 0x8, 0x7f7f20834e38, 0xc420d30e00, 0xc420dec700, 0x1485cbc33770374b, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3665 +0x426
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeWriteBatch(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0xc420b18510, 0x8, 0x1485cbc33770374b, 0x0, 0x100000001, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3457 +0x1e0
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).applyRaftCommandInBatch(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0xc420b18510, 0x8, 0x1485cbc33770374b, 0x0, 0x100000001, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3357 +0x127
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).evaluateProposal(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0x1, 0xc420b18510, 0x8, 0x100000001, 0x1, 0x1485cbc33770374b, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1872 +0x148
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).processRaftCommand(0xc42049dc00, 0x7f7f21a68000, 0xc420f43020, 0xc420b18510, 0x8, 0x23e, 0x1, 0x100000001, 0x1, 0xc420dec5b0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:3083 +0xa75
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).handleRaftReadyRaftMuLocked(0xc42049dc00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2338 +0x8a5
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).handleRaftReady(0xc42049dc00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2165 +0xcd
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processReady(0xc42033b180, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3193 +0x13e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42032e280, 0xc4203ecfc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:230 +0x28c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4203ecfc0, 0xc42019eda0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66```

--- a/pkg/roachpb/data.go
+++ b/pkg/roachpb/data.go
@@ -914,8 +914,8 @@ func (t Transaction) String() string {
 		fmt.Fprintf(&buf, "%q ", t.Name)
 	}
 	short := "<nil>"
-	if t.ID != nil {
-		short = t.ID.Short()
+	if id := t.ID; id != nil {
+		short = id.Short()
 	}
 	fmt.Fprintf(&buf, "id=%s key=%s rw=%t pri=%.8f iso=%s stat=%s epo=%d ts=%s orig=%s max=%s wto=%t rop=%t",
 		short, Key(t.Key), t.Writing, floatPri, t.Isolation, t.Status, t.Epoch, t.Timestamp,

--- a/pkg/storage/replica_command.go
+++ b/pkg/storage/replica_command.go
@@ -1413,9 +1413,16 @@ func (r *Replica) PushTxn(
 		if !pusherWins {
 			s = "failed to push"
 		}
+		pusherShort := "<nil>"
+		if id := args.PusherTxn.ID; id != nil {
+			pusherShort = id.Short()
+		}
+		pusheeShort := "<nil>"
+		if id := args.PusheeTxn.ID; id != nil {
+			pusheeShort = id.Short()
+		}
 		log.Infof(ctx, "%s "+s+" %s: %s (pushee last active: %s)",
-			args.PusherTxn.ID.Short(), reply.PusheeTxn.ID.Short(), reason,
-			reply.PusheeTxn.LastActive())
+			pusherShort, pusheeShort, reason, reply.PusheeTxn.LastActive())
 	}
 
 	if !pusherWins {

